std/node/process.ts

```diff
@@ -1,32 +1,109 @@
 // Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
 import { notImplemented } from "./_utils.ts";
+import EventEmitter from "./events.ts";
+import { fromFileUrl } from "../path/mod.ts";
+
+const notImplementedEvents = [
+  "beforeExit",
+  "disconnect",
+  "message",
+  "multipleResolves",
+  "rejectionHandled",
+  "SIGBREAK",
+  "SIGBUS",
+  "SIGFPE",
+  "SIGHUP",
+  "SIGILL",
+  "SIGINT",
+  "SIGSEGV",
+  "SIGTERM",
+  "SIGWINCH",
+  "uncaughtException",
+  "uncaughtExceptionMonitor",
+  "unhandledRejection",
+  "warning",
+];
 
 /** https://nodejs.org/api/process.html#process_process_arch */
 export const arch = Deno.build.arch;
 
+function getArguments() {
+  return [Deno.execPath(), fromFileUrl(Deno.mainModule), ...Deno.args];
+}
+
+//deno-lint-ignore ban-ts-comment
+//@ts-ignore
+const _argv: {
+  [Deno.customInspect]: () => string;
+  [key: number]: string;
+} = [];
+
+Object.defineProperty(_argv, Deno.customInspect, {
+  enumerable: false,
+  configurable: false,
+  get: function () {
+    return getArguments();
+  },
+});
+
+/**
+ * https://nodejs.org/api/process.html#process_process_argv
+ * Read permissions are required in order to get the executable route
+ * */
+export const argv: { [key: number]: string } = new Proxy(_argv, {
+  get(target, prop) {
+    if (prop === Deno.customInspect) {
+      return target[Deno.customInspect];
+    }
+    return getArguments()[prop as number];
+  },
+  ownKeys() {
+    return Reflect.ownKeys(getArguments());
+  },
+});
+
 /** https://nodejs.org/api/process.html#process_process_chdir_directory */
 export const chdir = Deno.chdir;
 
 /** https://nodejs.org/api/process.html#process_process_cwd */
 export const cwd = Deno.cwd;
 
-/** https://nodejs.org/api/process.html#process_process_exit_code */
-export const exit = Deno.exit;
+//deno-lint-ignore ban-ts-comment
+//@ts-ignore
+const _env: {
+  [Deno.customInspect]: () => string;
+} = {};
 
-/** https://nodejs.org/api/process.html#process_process_pid */
-export const pid = Deno.pid;
-
-/** https://nodejs.org/api/process.html#process_process_platform */
-export const platform = Deno.build.os === "windows" ? "win32" : Deno.build.os;
+Object.defineProperty(_env, Deno.customInspect, {
+  enumerable: false,
+  configurable: false,
+  get: function () {
+    return Deno.env.toObject();
+  },
+});
 
-/** https://nodejs.org/api/process.html#process_process_version */
-export const version = `v${Deno.version.deno}`;
+/**
+ * https://nodejs.org/api/process.html#process_process_env
+ * Requires env permissions
+ * */
+export const env: { [index: string]: string } = new Proxy(_env, {
```

Comment:
`Record<string, string>` ?

Reply:
Can you commit the change? I'm AFK right now